app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetControllerV2.kt

```diff
@@ -447,54 +453,73 @@ class BloodPressureEntrySheetControllerV2 @Inject constructor(
         .withLatestFrom(screenChanges)
         .filter { (_, screen) -> screen == DATE_ENTRY }
 
+    val prefilledDateStream = events
+        .ofType<BloodPressureDateToPrefillCalculated>()
+        .map { it.date }
+
     val newBpDataStream = Observables.combineLatest(dateValidations, bpValidations, patientUuidStream)
         .map { (dateResult, bpResult, patientUuid) ->
-          when {
-            dateResult is Valid && bpResult is Success -> SaveBpData.ReadyToCreate(dateResult.parsedDate, bpResult, patientUuid)
-            else -> SaveBpData.NeedsCorrection
+          if (dateResult is Valid && bpResult is Success) {
+            SaveBpData.ReadyToCreate(dateResult.parsedDate, bpResult.systolic, bpResult.diastolic, patientUuid)
+          } else {
+            SaveBpData.NeedsCorrection
           }
         }
 
     val updateBpDataStream = Observables.combineLatest(dateValidations, bpValidations, existingBpUuidStream)
         .map { (dateResult, bpResult, bpUuid) ->
-          when {
-            dateResult is Valid && bpResult is Success -> SaveBpData.ReadyToUpdate(dateResult.parsedDate, bpResult, bpUuid)
-            else -> SaveBpData.NeedsCorrection
+          if (dateResult is Valid && bpResult is Success) {
+            SaveBpData.ReadyToUpdate(dateResult.parsedDate, bpResult.systolic, bpResult.diastolic, bpUuid)
+          } else {
+            SaveBpData.NeedsCorrection
           }
         }
 
+    val toUtcInstant = { date: LocalDate ->
+      val userTime = LocalTime.now(userClock)
+      val userDateTime = date.atTime(userTime)
+      userDateTime.atZone(UTC).toInstant()
```

Comment:
I investigated why this still doesn't show the current date, and I figured out the reason. Let's discuss this more in person, but I'm going to leave the right code snippet to compute the `Instant` here for the record:

```kotlin
userDateTime
          .atZone(userClock.zone)
          .withZoneSameInstant(ZoneOffset.UTC)
          .toInstant()
```

Reply:
Fixed